Sources/SmithyJSON/JSONNode.swift

```diff
@@ -0,0 +1,21 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import class Foundation.NSNumber
+
+enum JSONNode: Equatable {
+    case bool(Bool)
+    case number(NSNumber)
+    case string(String)
+    case null
+    case array
+    case object
+}
```

Comment:
This enumeration represents the possible data types that are in the JSON object parsed from JSON data by the Foundation JSON parser.  The error below gets thrown when data cannot be fit into one of these enumeration cases.